### PR TITLE
JIP-4 - PostList + PostListItem oooooo

### PR DESCRIPTION
**Pull request by dryu99**
### Changes
- Implemented `PostList` and individual `PostListItem` components
- Updated return types for thunk actions
- Renamed some files to be more consistent / make more sense / hmmmmm
- Allow user to vote for posts (needs more work tho since only logged in users need to vote, need to keep track of what posts user have voted for, etc)
- Implement basic  NEW / POPULAR filtering (have to collectively agree on how we want to handle post fetching/filtering tho)
- Add `styled-components` theme provider so we can access theme props in `styled-components` components 
- Prob sth else 

I'll use storybook next time I promise AAAAA

Screen Shot 2021-06-12 at 12 11 15 AM

### CODE REVIEW

File: packages/client/src/pages/example/example.tsx [hunk 1]
```diff
@@ -167,12 +167,19 @@ const PostExample = () => {
   const vancouver: Location = { name: "Vancouver", lat: 49.26, lon: -123.22 };
   const surrey: Location = { name: "Surry", lat: 49.26, lon: -123.22 };
   const richmond: Location = { name: "Richmond", lat: 49.26, lon: -123.22 };
-  const yellowknife: Location = { name: "Yellowknife", lat: 49.26, lon: -123.22 };
+  const yellowknife: Location = {
+    name: "Yellowknife",
+    lat: 49.26,
+    lon: -123.22,
+  };
```
Comment (srhsrhsrhsrh) [inline packages/client/src/pages/example/example.tsx:174]: did u change this formatting on purpose to emphasize that ur past wrongs are righted

**Review by srhsrhsrhsrh — APPROVED** [review]
cheesy slices

File: packages/client/src/pages/example/example.tsx (hunk 1 above)
Comment (dryu99) [reply packages/client/src/pages/example/example.tsx:174]: the good ppl of yellowknife deserve it

File: packages/client/src/services/users.ts [hunk 2]
```diff
@@ -30,9 +34,15 @@ const login = async (credentials: LoginCredentials) => {
...
 
+const update = async (id: string, partialUser: Partial<UserState>) => {
+  const response = await axios.put(`/users/${id}`, partialUser);
+  return response;
+};
+
 const authService = {
   signup,
   login,
+  update,
```
Comment (danyliu9) [inline packages/client/src/services/users.ts:45]: Could we change this to updateUser please? 😊 And like maybe signUp so we stay consistent with Camel case? Smol ting but good ting y'knowmsayn?

**Review by danyliu9 — COMMENTED** [review]
I sure hope that guy doesn't find any toes..

File: packages/client/src/services/users.ts (hunk 2 above)
Comment (dryu99) [reply packages/client/src/services/users.ts:45]: good point my brain fry, i will update update in the next task

**Pull request merged**